util/explain_analyze_helper.sql

```diff
@@ -0,0 +1,46 @@
+CREATE OR REPLACE FUNCTION pganalyze.explain_analyze(query text, params text[], param_types text[], analyze_flags text[]) RETURNS text AS $$
+DECLARE
+  prepared_query text;
+  params_str text;
+  param_types_str text;
+  explain_prefix text;
+  explain_flag text;
+  result text;
+BEGIN
+  SET TRANSACTION READ ONLY;
+
+  PERFORM 1 FROM pg_roles WHERE (rolname = current_user AND rolsuper) OR (pg_has_role(oid, 'MEMBER') AND rolname IN ('rds_superuser', 'azure_pg_admin', 'cloudsqlsuperuser'));
+  IF FOUND THEN
+    RAISE EXCEPTION 'cannot run: pganalyze.explain_analyze helper function is owned by superuser - recreate function with lesser privileged user';
+  END IF;
+
+  SELECT pg_catalog.regexp_replace(query, ';+\s*\Z', '') INTO prepared_query;
+  IF prepared_query LIKE '%;%' THEN
+    RAISE EXCEPTION 'cannot run pganalyze.explain_analyze helper function when query contains semicolon';
```

Comment:
Also, how does this handle a single query that ends with a semicolon? I'd expect that to be allowed, since there isn't a statement following the semicolon.

Reply:
The single query ending in a semicolon is handled by the `regexp_replace` in the prior line.

I'd argue this check is essential - if we don't do it, the whole purpose of having a function to turn a read query into the EXPLAIN ANALYZE output only would be defeated, since you could just run arbitrary queries. Unless you have another idea of how to protect against this at the SQL level? (as mentioned previously, we really do need multiple layers of defense, and a function is much easier to audit than all the collector's Go code).

Reply:
Regarding the earlier feedback, I removed the function name from the messages (since you're right, it would be contained in the CONTEXT as part of error messages), and also reworded the semicolon error to instead refer to multi-statement queries.

Reply:
After testing this locally, it looks like the `CONTEXT` doesn't make it to the server as part of the query run's error string. So maybe the function name should be added back?

Reply:
Ah, yeah, you're right. I'll add it back for now to make it clearer for the user.